_posts/2025-05-07-release-notes-3.7.0.md

```diff
@@ -0,0 +1,452 @@
+
+Preview language features and APIs are guaranteed to be standardized in some next Scala minor release, but allow the compiler team to introduce small, possibly binary incompatible, changes based on the community feedback.
+These can be used by early adopters who can accept the possibility of binary compatibility breakage. For instance, preview features could be used in some internal tool or application. On the other hand, preview features are discouraged in publicly available libraries.
+
+All preview features can be enabled using `-preview` flag passed to the compiler. More information about preview features can be found in [preview definitions guide](https://docs.scala-lang.org/scala3/reference/preview/overview.html)
+
+### [SIP-62: For comprehension improvements](https://docs.scala-lang.org/sips/better-fors.html)
+
+The first introduced preview feature is [SIP-62 - `Better Fors`](https://docs.scala-lang.org/sips/better-fors.html) originally introduced in Scala 3.6 as an experimental.
+A major user-facing improvement introduced by this SIP is the ability to start a for-comprehension block with aliases.
+It also introduces changes to how your code is desugared by the compiler, leading to a more optimized code by removing some redundant calls.
+
+```scala
+//> using options -preview
+//> using scala 3.7
+@main def betterFors = 
+  for
+    a = 1
+    b <- Some(2)
+    c <- Option.when(a < 5)(10)
+  yield b * c
+```
+
+We encourage users to try out this feature which is likely going to be stabilised in Scala 3.8.
+
+## New experimental SIPs
+
+### [SIP-61: Unroll Default Arguments for Binary Compatibility](https://docs.scala-lang.org/sips/unroll-default-arguments.html)
+
+Scala case classes were for a long time avoided when defining a public API since each change to the list of their fields affected their binary compatibility. This was typically observed whenever a new field was added to the case class primary constructor, even if it was defined with a default value to resolve source compatibility problems. Even though binary compatibility issues were typically detected when using [MiMa](https://github.com/lightbend/mima), you were still required to manually create shim methods to match the old signatures of the methods.
+
+The `@unroll` annotation is going to eliminate this long-standing issue. This annotation lets you add additional parameters to method `def`s, `class` constructors, or `case class`es, without breaking binary compatibility. `@unroll` works by generating “unrolled” or “telescoping” forwarders, ensuring your code will maintain binary compatibility as new default parameters and fields are added over time.
+
+```scala
+//> using scala 3.7
+//> using options -experimental
+
+import scala.annotation.unroll
+
+case class Product(
+    name: String,
+    price: BigDecimal,
+    @unroll description: Option[String] = None,
+){
+  // generated by compiler:
+  // def this(name: String, price: BigDecimal) = this(name, price)
```

Comment:
maybe add a placeholder to show that it is using the default argument in the forwarder, otherwise it looks recursive